scipy/sparse/construct.py

```diff
@@ -405,15 +405,17 @@ def _compressed_sparse_stack(blocks, axis):
     other_axis = 1 if axis == 0 else 0
     data = np.concatenate([b.data for b in blocks])
     indices = np.concatenate([b.indices for b in blocks])
+    idx_dtype = get_index_dtype(arrays=[b.indptr for b in blocks],
+                                maxval=data.size)
     indptr = []
-    last_indptr = 0
+    last_indptr = idx_dtype(0)
     constant_dim = blocks[0].shape[other_axis]
     sum_dim = 0
     for b in blocks:
         if b.shape[other_axis] != constant_dim:
             raise ValueError('incompatible dimensions for axis %d' % other_axis)
         sum_dim += b.shape[axis]
-        indptr.append(b.indptr[:-1] + last_indptr)
```

Comment:
I see, that is a good point. However, this also changes the dtype in the original matrices. If, as you write above, indices and indptr have to be the same dtype, those original matrices may become damaged (i.e do we have to change the dtype in the input matrices as well?).